src/operator/contrib/nms.cc

```diff
@@ -0,0 +1,55 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file nms.cc
+ * \brief Non-maximum suppression for bounding boxes, cpu operator
+ * \author Joshua Zhang
+ */
+#include "./nms-inl.h"
+#include "../elemwise_op_common.h"
+
+namespace mxnet {
+namespace op {
+DMLC_REGISTER_PARAMETER(NMSParam);
+
+NNVM_REGISTER_OP(_contrib_nms)
+.add_alias("_contrib_non_maximum_suppression")
+.describe(R"code(Apply non-maximum suppression to input.
+
+The output will be sorted in descending order according to `score`. Boxes with
+overlaps larger than `overlap_thresh` will be removed, and the corresponding
+position will be recorded for backward propogation.
```

Comment:
What does this mean? How is gradient for removed boxes handled? What are the -1s doing?